src/main.rs

```diff
@@ -260,53 +266,74 @@ fn exec(cmd: Command) -> Result<String> {
         Command::Build {
             manifest_path,
             verbosity,
+            skip_bundle,
+            skip_metadata,
             unstable_options,
         } => {
+            if *(skip_metadata) {
+                let manifest_path = ManifestPath::try_from(manifest_path.as_ref())?;
+                let dest_wasm = cmd::build::execute(
+                    &manifest_path,
+                    verbosity.try_into()?,
+                    unstable_options.try_into()?,
+                    true,
+                )?;
+                return Ok(format!(
+                    "\nYour contract's code is ready. You can find it here:\n{}",
+                    dest_wasm.display().to_string().bold()
+                ));
+            }
             let manifest_path = ManifestPath::try_from(manifest_path.as_ref())?;
-            let dest_wasm = cmd::build::execute(
+            let metadata_result = cmd::metadata::execute(
                 &manifest_path,
                 verbosity.try_into()?,
+                false,
                 unstable_options.try_into()?,
             )?;
-            Ok(format!(
-                "\nYour contract is ready. You can find it here:\n{}",
-                dest_wasm.display().to_string().bold()
-            ))
-        }
-        Command::Pack {
-            manifest_path,
-            verbosity,
-            unstable_options,
-        } => {
-            let manifest_path = ManifestPath::try_from(manifest_path.as_ref())?;
-            let pack_file = cmd::metadata::execute(
-                manifest_path,
+            if *(skip_bundle) {
+                return Ok(format!(
+                    "\nYour contract's code is ready. You can find it here:\n{}
+                    \nYour contract's metadata is ready. You can find it here:\n{}",
+                    metadata_result.wasm_file.display().to_string().bold(),
+                    metadata_result.metadata_file.display().to_string().bold(),
+                ));
+            }
+            let bundle_result = cmd::metadata::execute(
+                &manifest_path,
```

Comment:
Please move as much of the logic as possible into `cmd/build.rs`.

Even better see https://github.com/paritytech/cargo-contract/pull/79 where I have extracted command variants into their own struct e.g. https://github.com/paritytech/cargo-contract/blob/5d5f7bee958e403e654ce4f3dda92db4c8757762/src/cmd/extrinsics/call.rs

Goal is to make `main.rs` as slim as possible eventually just dispatching the commands.